transcom/mymove
# B-17908-INT KICKBACK (On pause, awaiting PO)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-17908)

## Summary

PO requested to capture initial PPM shipment submission in the Move history logs.

### How to test

1. Create a PPM shipment as a customer and submit your move
2. Login as an office user and access the move. Navigate to move history and confirm that the submission is populated

Note: you will see Updated shipment: Status: SUBMITTED twice, as we track changes to the status of both mto_shipments and ppm_shipments.

NEW TEST 3/13/24
It was requested that we make a custom event "PPM status: ADDED" and tie it to the completion of a customer submitted a PPM. To keep this a low level of effort, I tied it to the requesting of an advance, as that is the last screen a customer sees when creating a PPM. The only issue is, if the incentive is reset (by editing the PPM), we will see "ADDED" again the next time the advance is dealt with. This was deemed ok by the PO.

1. Create a PPM shipment as a customer and don't request an advance
2. Login as an QAE office user and find the move. Check the move history and confirm you see an entry for PPM status: ADDED
![image](https://github.com/transcom/mymove/assets/147535220/8af7676c-3387-4913-a5c0-9f54db3a8dd6)
3. As a customer, edit the PPM, don't change any details, and advance to the advance request screen. Request an advance and confirm that the requested advance does not include the PPM status: ADDED
![image](https://github.com/transcom/mymove/assets/147535220/c2cd60c5-f019-43d2-9919-1bc026fc4e97)
